Automattic/wp-calypso
# A4A: Removing the A4A environments from isJetpackCloud, adding the A4A environment check anywhere that it is needed

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/242

## Proposed Changes

* This PR removes the A4A environments from `client/lib/jetpack/is-jetpack-cloud.ts`. It also adds checks for the A4A environment with the newly created `isA8CForAgencies`, anywhere `isJetpackCloud` is used that appears relevant.
* This also includes a couple of small style tweaks as button styles in particular have been affected (largely due to the removal of the `is-jetpack-site` class).

What may need looking at later:
* We may need to create tests here that cover A4A aswell: `components/notes-formatted-block/test/blocks.js`. 
* We do not have an A4A platform yet to fetch eg. sites per user from:`components/site-selector/utils.ts`.
* `my-sites/controller.js` may need updating at some point when we have a plugins path, non JP cloud specific site selection considerations, and primary site needs.

There are references to `isJetpackCloud` that I've left adding checks for the a4a environment alongside. These predominantly include:
- We don't seem to use `VisibleDaysLimitUpsell`
- Jetpack social paths
- `JetpackComMasterbar` as we're not using it
- `Jetpackconnect`
- GA cross domain linking: `/lib/analytics/ga.js`
- Hotjar site settings: `/lib/analytics/hotjar.js`
- Other tracking: `/analytics/tracker-buckets.ts`, `analytics/ad-tracking/load-tracking-scripts.js`, `analytics/ad-tracking/setup.js`, `analytics/utils/get-ga-gtag.ts`
- `settingsBasePath` - we don't have one yet for A4A
- `PlanBillingPeriod` and `PaymentMethodBackupToggle`
- `/my-sites/backup/controller.js` - we have our own a4a version of this file
- Some checks in `/my-sites/backup/main.jsx` left as the current style relies on them
- Checkout thank you page URL redirect: `/my-sites/checkout/get-thank-you-page-url/index.ts`
- `/earn` path: `/my-sites/earn/controller.ts` and earn related files / components.
- `JetpackSearchContent` and anything related to Jetpack Search
- `/marketing/connections` related
- const `isJetpackSiteOrJetpackCloud` (sitePicker related)
- `/my-sites/plans/jetpack-plans/` related (we have our own license checkout process)
- `/my-sites/purchases` related
- `const notFoundIfNotEnabled` - this has not been modified for my-sites/backup and scan, as there is a commented note in this A4A file about it not working in A4A: `/a8c-for-agencies/sections/sites/features/jetpack/backup/routes.ts`
- `/my-sites/sidebar`
- `/my-sites/sites` (upgrade related)
- `/my-sites/subscribers`

## Testing Instructions

For the below testing steps, make sure to run `yarn start-a8c-for-agencies` locally and then visit `http://agencies.localhost:3000/sites`.

Some of the changes have been difficult to test, but the below is testable:

* Test the preview pane Activity feed, eg. plugins and themes (and author) links will now be removed.
* Some classnames have been changed (eg. removing `wordpressdotcom` as an additional class in some cases) - so checking general styles aren't impacted would be good. Specifically: activity preview pane.
* In the preview pane, for any site under Activity, Backup, Scan that shows a time mismatch warning, the 'update' link should continue taking you to `wp-admin`, not `/settings/general/`
* Other things to test generally include Scan and Backup tabs in the preview pane (upsells, links, styles)
* Generally test functionality on the A4A dashboard to make sure nothing is broken. 
* Ideally also check Jetpack Cloud and WordPress.com environments (in particular with relation to Scan, Backup, Activity , Add new site buttons)

## Pre-merge Checklist
